src/cow.jl

```diff
@@ -71,117 +93,125 @@ maybe_copy(x::AbstractDict) = begin
 end
 
 # (func_name => mutated_arg_pos)
-const MUTATING_OPS = Dict{Symbol, Int}(
-    :setindex! => 1,
-    :push! => 1, :pushfirst! => 1,
-    :pop! => 1, :popfirst! => 1,
-    :append! => 1,
-    :delete! => 1, :deleteat! => 1,
-    :setdiff! => 1,
+const MUTATING_OPS = Dict{Any, Vector{Int}}(
+    :setindex! => [1],
+    :push! => [1], :pushfirst! => [1],
+    :pop! => [1], :popfirst! => [1],
+    :append! => [1],
+    :delete! => [1], :deleteat! => [1],
+    :setdiff! => [1],
 )
 
 """
     mutating(e)
 
 Predicate if an expression is a mutating operation.
 If true, returns `ture` and the position of the mutated argument;
-else returns `(false, 0)`.
+else returns `(false, [0])`.
 
 Examples:
 
-- `push!(a, v)` -> (true, 1)
-- `print(a)` -> (false, 0)
+- `push!(a, v)` -> (true, [1])
+- `print(a)` -> (false, [0])
 
 """
 function mutating(e)
-    isa(e, Expr) || return (false, 0)
-    e.head == :call || return (false, 0)
-    if isa(e.args[1], GlobalRef) && haskey(MUTATING_OPS, e.args[1].name)
-        return (true, MUTATING_OPS[e.args[1].name])
+    isa(e, Expr) || return (false, [0])
+    e.head == :call || return (false, [0])
```

Comment:
You could use `Meta.isexpr(expr, :call)` and an `if` block around the whole inner code, instead.

Reply:
Or alternatively just
```suggestion
    Meta.isexpr(expr, :call) || return (false, [0])
```
if you don't want to wrap everything in an `if` block.